examples/gno.land/r/malek_/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,218 @@
+package gnofundme
+
+import (
+	"bytes"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+	pusers "gno.land/p/demo/users"
+	gfm "gno.land/p/malek_/gnofundme" // GnoFundMe package
+	"gno.land/r/demo/users"
+)
+
+var (
+	campaigns      *avl.Tree
+	banker         std.Banker
+	balance        int64
+	campaignPrefix = "/r/demo/gnofundme:"
+	campaignsTotal = make(map[string]int64)
+)
+
+func init() {
+	campaigns = avl.NewTree()
+	banker = std.GetBanker(std.BankerTypeRealmIssue)
+}
+
+func NewCampaign(ct int, title string, description string, goal uint64, beginT int64, deadlineT int64, beneficiary std.Address) string {
+	if ct < 0 || ct > 5 {
+		panic("invalid campaign type")
+	}
+
+	var beneficiaryU *pusers.User
+	// gets the caller address (owner of the campaign)
+	caller := std.PrevRealm().Addr()
+
+	// verify that the user owner exists
+	owner := users.GetUserByAddress(caller)
+	if owner == nil {
+		panic("owner not registered in 'gno.land/r/demo/users'")
+	}
+
+	// verify that the benefeciary exists if it's different from caller
+	if beneficiary != caller {
+		beneficiaryU = users.GetUserByAddress(beneficiary)
+		if beneficiaryU == nil {
+			panic("beneficiary not registered in 'gno.land/r/demo/users'")
+		}
+	} else {
+		beneficiaryU = owner
+	}
+
+	// Verfiy that the dates are correct
+	if beginT < time.Now().Unix() {
+		panic("begin date must be in the future")
+	}
+	if deadlineT <= beginT {
+		panic("deadline date must be after begin date")
+	}
+
+	// creates the time objects
+	begin := time.Unix(beginT, 0)
+	deadline := time.Unix(deadlineT, 0)
+	// create the campaign
+	campaign := gfm.NewCampaign(ct, title, description, goal, begin, deadline, owner, beneficiaryU)
+
+	// the key will be the size of the tree
+	key := campaigns.Size()
+	campaigns.Set(strconv.Itoa(key), campaign)
+	std.Emit("Campaign Creation", "campaign_created", campaign.CampaignToJSONString())
+	return "Campaign N°" + strconv.Itoa(key) + " created successfully"
+}
+
+func AddContributor(campaignId string, display bool) string {
```

Comment:
Shouldn't this function be named `Donate` or `Contribute`? The people giving the money will be calling it.